packages/yjs/src/lib/BaseYjsPlugin.ts

```diff
@@ -1,33 +1,91 @@
 import type { HocuspocusProviderConfiguration } from '@hocuspocus/provider';
 import type { WithCursorsOptions } from '@slate-yjs/core';
+import type { Awareness } from 'y-protocols/awareness';
+import type { WebrtcProvider } from 'y-webrtc';
 
 import { HocuspocusProvider } from '@hocuspocus/provider';
 import { type PluginConfig, createTSlatePlugin } from '@udecode/plate';
+import { WebrtcProvider as YWebrtcProvider } from 'y-webrtc';
+import * as Y from 'yjs';
 
 import type { WithYjsOptions } from './withTYjs';
 
 import { withPlateYjs } from './withPlateYjs';
 
+export interface UnifiedProvider {
+  awareness: Awareness;
+  connect: () => void;
+  destroy: () => void;
+  disconnect: () => void;
+  getDocument: () => Y.Doc;
+}
+
+export type WebRTCProviderOptions = {
+  /** Room name for the collaboration */
+  roomName: string;
+  /** Whether to filter broadcast channel connections */
+  filterBcConns?: boolean;
+  /** Maximum number of WebRTC connections */
+  maxConns?: number;
+  /** Optional password for the room */
+  password?: string;
+  /** Additional peer options for simple-peer */
+  peerOpts?: Record<string, unknown>;
+  /** Optional signaling servers. Defaults to public servers if not specified */
+  signaling?: string[];
+};
+
 export type YjsConfig = PluginConfig<
   'yjs',
   {
     isConnected: boolean;
     isSynced: boolean;
-    provider: HocuspocusProvider;
+    provider: UnifiedProvider;
     /** WithCursors options */
     cursorOptions?: WithCursorsOptions;
     disableCursors?: boolean;
-    /**
-     * HocuspocusProvider configuration
-     *
-     * @required
-     */
+    /** HocuspocusProvider configuration Required if providerType is 'hocuspocus' */
     hocuspocusProviderOptions?: HocuspocusProviderConfiguration;
+    /** Provider type - defaults to hocuspocus for backward compatibility */
+    providerType?: YjsProviderType;
```

Comment:
What about removing this and derive the type from which one is defined between `hocuspocusProviderOptions` and `webrtcProviderOptions`? Then we could store add the type in `provider.type`.